### DISCUSSION TITLE
max. sprite size for DMA on Pico

### DISCUSSION THREAD

**Discussion by elral** [Bodmer/TFT_eSPI#2117] [Raspberry Pi Pico]
I went through your examples using the sprites on a Pico an got some running. I also transferred your Artificial Horizon example from the Arduino Forum to use with sprites (where I am wonderung what the best way will be to make it round, some ideas but maybe I will ask again ;) )
The only topic I am wondering, I read in your comments that the max. size for DMA transfer is 64kB. But setting up a 240 * 320 display in 2 buffers will result in 75kB per Buffer.
Is the limit of 64kB not valid for the Pico?
If so, what would be the max. value?
And wouldn't it make sense on this case only to use one sprite buffer to avoid double drawing?

Not to forget, thanks for this excellent, powerfull and fast library.

**Comment by Bodmer** [comment]
The limit is for an ESP32, for a RP2040 processor the limit will be the RAM size.

One sprite buffer can be used. The advantage of double buffering is that the processor can prepare the next buffer while the previous one is being sent to the screen by the DMA controller.
